core/src/avm2/domain.rs

```diff
@@ -245,6 +245,9 @@ impl<'gc> Domain<'gc> {
     }
 
     pub fn export_class(&self, class: GcCell<'gc, Class<'gc>>, mc: MutationContext<'gc, '_>) {
```

Comment:
(Nit: Maybe refactor the key value into its own variable, like:

```rust
        let class_name = class.read().name();
        if self.0.read().classes.contains_key(class_name) {
            return;
        }
        self.0.write(mc).classes.insert(class_name, class);
```

?)

Reply:
I think it's readable enough as-is.